pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,120 @@ package sdkstats
-			if err != nil {
-				return nil, err
-			}
+//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
+func GetStats(client *github.Client, rq RepoMetadataQuery) ([]RepoMetadata, error) {
+	var repoList []RepoMetadata
+	var sdkVersion string
+	var skipVersion int
 
-			repoDetails := &RepoMetadata{
-				FullName:     fullName,
-				ProjectType:  projectType,
-				SDKVersion:   sdkVersion,
-				Stars:        Stars,
-				CreatedAt:    repoMap["CreatedAt"],
-				PushedAt:     repoMap["PushedAt"],
-				TotalCommits: TotalCommits,
+	for _, q := range rq.Queries {
+		searchOp, err := Search(client, q)
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(q, "Gopkg.toml") {
+			skipVersion = 1
+		}
+		for j := 0; j < len(searchOp); j++ {
+			for i := 0; i < len(searchOp[j].CodeResults); i++ {
+				if skipVersion == 1 {
+					sdkVersion = "N/A"
+				} else {
+					sdkVersion, err = rq.VersionParser.ParseVersion(searchOp[j].CodeResults[i], rq.ProjectType)
+					if err != nil {
+						return nil, err
+					}
+				}
+				repoDetails := &RepoMetadata{
+					URL:         searchOp[j].CodeResults[i].GetRepository().GetURL(),
+					Name:        searchOp[j].CodeResults[i].GetRepository().GetName(),
+					Owner:       searchOp[j].CodeResults[i].GetRepository().GetOwner().GetLogin(),
+					ProjectType: rq.ProjectType,
+					SDKVersion:  sdkVersion,
+				}
+				repoDetails, err = GetRepoDetails(client, repoDetails)
+				if err != nil {
+					return nil, err
+				}
+				repoList = append(repoList, *repoDetails)
 			}
-			repoList = append(repoList, *repoDetails)
-
 		}
 	}
-
 	return repoList, nil
 }
 
 //Search returns Github API results for Code Search, for a given Query string
 func Search(client *github.Client, q string) ([]*github.CodeSearchResult, error) {
-
-	var searchOp []*github.CodeSearchResult
+	var searchop []*github.CodeSearchResult
 	ctx := context.Background()
-
 	opts := &github.SearchOptions{TextMatch: true,
 		ListOptions: github.ListOptions{
 			PerPage: 100,
 		},
 	}
+	//Below Logic enables to collate Search results from all pages returned from API call.
 	for {
 		op, resp, err := client.Search.Code(ctx, q, opts)
 		if err != nil {
 			return nil, err
 		}
-		searchOp = append(searchOp, op)
+		searchop = append(searchop, op)
 		if resp.NextPage == 0 {
 			break
 		}
 		opts.Page = resp.NextPage
 	}
-	return searchOp, nil
-
+	return searchop, nil
 }
 
-//GetRepoDetails Returns map[string][string] with Stars,CreatedAt,PushedAt, and TotalCommits details for a given onwer, and repo
-func GetRepoDetails(client *github.Client, owner string, name string) map[string]string {
-
-	repoMap := map[string]string{}
-
+//GetRepoDetails Returns []RepoMetaData with Stars,CreatedAt,PushedAt, and TotalCommits details for a given owner, and repo
+func GetRepoDetails(client *github.Client, repoDetails *RepoMetadata) (*RepoMetadata, error) {
 	ctx := context.Background()
```

Comment:
In library code, it's best to let users pass their own context. The convention is to make the first argument in exported functions be the context, so this would be:

```go
func GetRepoDetails(ctx context.Context, client *github.Client, repoDetails *RepoMetadata) (*RepoMetadata, error) {
```

Since we call `GetRepoDetails` from another function in this library, we should also update _that_ function's signature as well, and so on, until we reach a function that is not called anywhere else.

Reply:
Will do.